### ISSUE TITLE
Module resolution should be done before filtering for unique modules

### ISSUE THREAD

**Issue by phated** [gulpjs/evented-require#2]
I think there's an issue with the unique filtering implementation where you could do something like `'./foo.js'` and `../fixtures/foo.js` and both would get loaded.  I think `requireAll` could have a better implementation to handle cases like this.

**Comment by sttk** [comment]
That's friendly for logging.
On the other hand, if an user specified 3 module paths but he got only 2 result, he may wonder it. Or a program may expect to get pairs of module path and result.

**Comment by phated** [comment]
@sttk Does that mean I should remove the unique filter from this module and leave it in Liftoff?

**Comment by sttk** [comment]
@phated I just suggested possible requirements or issues. I think there are three ways for this: 1. use unique filtering, 2. remove unique filtering, 3. select 1 or 2 by options, but I cannot determine which is the best way.

**Comment by phated** [comment]
I think we should remove it and keep the simple filtering within Liftoff.  I was just trying to simplify things but I don't like the feature being poorly implemented here.

**Comment by phated** [comment]
Closing in favor of #5
